### ISSUE TITLE
iOS - App review rejected

### ISSUE THREAD

**Issue by Shaninnik** [WhatsApp/stickers#59]
Not really a code issue, but I thought that I might let you know: Just followed your guide for iOS and app review was rejected for several reasons: 

Guideline 2.3.3 - Performance - Accurate Metadata
I have included only one screenshot of the main screen of application, but they decided that it is not enough. Almost sure that I can't include screenshots of WhatsApp so not sure what else can I do here.

Guidelines 2.3.7 - Performance - Accurate Metadata
My app name has `WhatsApp` as part of it's name because I thought that it is appropriate to use it here because it is designed specifically for WhatsApp, and does not do anything on it's own. Apple review team thinks another way.

Guideline 4.2 - Design - Minimum Functionality
This is the most serious reason they gave me. They have found that usefulness of this app is limited...

I have submitted an appeal and waiting for their answer. It might be good to update readme to cover some of this topics, I am sure that I will not be the only one to be rejected.

**Comment by SuperSandro2000** [comment]
Guideline 2.3.3:
I would suggest to include a screen of the emoji selection where you add the to whatsapp and the info screen.

2.3.7
I did a quick search for sticker for whatsapp on the app store and found plenty. not sure about that.

4.2
If you only add like 5 sticker then I could agree with them but there is also trash on the app store that isn't useful in any way. Maybe add some more stickers?

**Comment by Shaninnik** [comment]
Yeah I can include Info screen I guess. Do you know if they will object if I will include screenshots from WhatsApp itself with my stickers loaded?

4.2 - 17 top-quality stickers, had them as Messages app for quite a long a time (https://itunes.apple.com/us/app/enjoy-the-cuteness-stickers/id1350136791?ls=1&mt=8). I think my reviewer have not liked the whole concept of the app that does almost nothing.

**Comment by SuperSandro2000** [comment]
I just gave you general advice as I never listed an app on the play store but I saw other apps doing the same so I would assume it is fine.

I could imagine that. Just ask friendly and try it again.

**Comment by Shaninnik** [comment]
Not the first time I am doing an appeal to app review so I am sure it will be fine after some trial and error. I just thought that since you already provide detailed guide of what to do, what can be included and what not, that I've just followed, you may have direct solutions to resolve my issues, or some experience with review process of apps that are based on your code. Maybe some info from me can help in improving readme file.

**Comment by podycust** [comment]
Same here!

**Comment by justinheintz** [comment]
I’ve been rejected...

Guideline 4.3 - Design

We noticed that your app provides the same feature set as other apps submitted to the App Store; it simply varies in content or language, which is considered a form of spam.

The next submission of this app may require a longer review time, and this app will not be eligible for an expedited review until this issue is resolved.

**Comment by Shaninnik** [comment]
Yep. Received same response after submitting an appeal and writing them in details what my app do. It seems that it is almost impossible to pass app review if you are using sample code

**Comment by knothed** [comment]
Same issue here.

**Comment by sguidobono** [comment]
I have the same problem and I do not know how I can solve it. If anyone knows what can be done, it would be great.

**Comment by ericcheungwork** [comment]
I got the same reply from apple today:

---

Guideline 4.3 - Design

We noticed that your app provides the same feature set as other apps submitted to the App Store; it simply varies in content or language, which is considered a form of spam.

The next submission of this app may require a longer review time, and this app will not be eligible for an expedited review until this issue is resolved.

**Comment by abarreda** [comment]
I got also same reply from Apple.

> Guideline 4.3 - Design
> 
> We noticed that your app provides the same feature set as other apps submitted to the App Store; it simply varies in content or language, which is considered a form of spam.
> 
> The next submission of this app may require a longer review time, and this app will not be eligible for an expedited review until this issue is resolved.

It seems they don't want to publish stickers for WhatsApp, other than the ones that are already published.

**Comment by ptkdev** [comment]
Same error of **Guideline 4.3 - Design**, apple publish my [version 1.0](https://itunes.apple.com/app/ptkdev-stickers-for-wa/id1440418846) but not update 1.1

**Comment by firstangel999** [comment]
ok i'm ready to send to review the next update of my stickers app. Graphically revamped and added a news section, an help guide and push notification
Stay tuned for "iRejected" Season 2, a grotesque comedy by Apple!

**Comment by sguidobono** [comment]
I just talked on the phone with an Apple app store representative.
And they told me that it does not matter if there are other applications that contain "Stickers for WhatsApp" in their title. Or use the same code.
They will never accept my application with this code.
Because there are already similar applications and because adding WhatsApp stickers is a very unusable functionality for an application.
They told me that I should change the concept of it.
Clearly WhatsApp did not contemplate this when it shared the code. We will have to wait.

**Comment by firstangel999** [comment]
with this way of thinking, stickers packs for imessage are also useless

**Comment by ptkdev** [comment]
Is impossibile update my stickers app. Apple reject my app 4 times. Nice work, this shit method not work for publish stickers. Apple has idiots tester.

**Comment by knothed** [comment]
Any updates on this?

**Comment by cartoonsmart** [comment]
Hey guys, I've talked with Apple Review at length about stickers in general. Bottom-line, more than one sticker pack per developer is one too many. I had around 200 at one point, now its down to one container pack. I tried hard to keep just a handful of them (especially since those particular ones had a solid user base). 

This isn't about one particular reviewer not approving something (or not listening to you), it's a total restructuring of the App Store. I've got a few games in the store as well, and I'm just waiting for the eventual Connect message that says they are Design Spam.  And if that day comes, I've got to decide if its worth my time to combine those games into a single container app, or just say, screw it, its not worth it. If I decide the latter, well, that sucks for those buyers that one day want to restore those games, and it sucks I spent a lot of time way-back-when building those games, but the end decision was on me to not make a container. 

So regarding WhatsApp... Apple isn't going to re-flood the store like they did in 2016 with sticker packs. If they won't do it for Messages stickers, they sure aren't going to for WhatApp stickers.

**Comment by MarsVard** [comment]
@cartoonsmart I had a single sticker pack app, it got removed this morning. 🤷‍♂️

**Comment by cartoonsmart** [comment]
Dang, that is rough. I figured your best chance would be if it was your one and only app... You've probably already done this, but if you search 4.3 in the Apple Dev forums, you'll find plenty of examples of this (and it's been going on for at least a year or more).... I feel for all of ya. It's taken me a long time to accept it myself. 

I will say reading the size limits on these WhatsApp stickers, at 100kb max, that isn't much of a playground to work with. I'm not sure if your stickers were animated, but it might be worth asking your reviewer if there was a quality factor involved in the decision. For example, if you had animated them and maybe included x amount, if that would pass review. I've seen plenty of 2-frame animations that were a lot more exciting than static images.  A little goes a long way.

**Comment by firstangel999** [comment]
@cartoonsmart in the end the only solution is to create a single stickers app?

**Comment by cartoonsmart** [comment]
Exactly. You'll need to tell App Review which will be your "container app", and that app you get to keep in the store (so obviously, keep your most profitable or most downloaded).... They also recommended before I submit that container app, that I had already toggled Remove from Sale on all the other packs. 

Obviously if you've only been using the Xcode Sticker template, its impossible to keep using that as a container. Plus, you'll need to convert your frames to APNGs (tedious), and then create an iMessage application with multiple upsell options for separate packs. If thats the route you want to go. 

You could also make a mega pack of everything. Just stuff them into one pack.  Kittens meet Zombies meet Santa.

**Comment by firstangel999** [comment]
@cartoonsmart no fortunately i don't use the standard template for iMessage stickers in the next build (the 86758757 build) of my stickers app....i have made a iMessage app with stickers divided in categories etc etc. I upload it soon, shutdown the oldest and i hope to be not rejected another time. Thank you for the advice :)

**Comment by cartoonsmart** [comment]
No problem, glad to help! And yeah, give them a peace offering, shutdown the old stuff first and then submit. From what I can tell removing from sale isn't permanent anyway, but I'm guessing Apple wouldn't be too happy if you tried that in the future!  Don't poke the bear =)

**Comment by firstangel999** [comment]
@cartoonsmart at the end i don't have interest to re-put in sell removed stickers pack. The bear scares me lol

**Comment by ptkdev** [comment]
@cartoonsmart where you find this information of "container app"? I create single app stickers imessage+whatsapp but don't pass review... =/

**Comment by cartoonsmart** [comment]
They called me initially about this.... Is that your only app in the store?

**Comment by cartoonsmart** [comment]
**Pure speculation here...** I do wonder if they've soured a bit on approving sticker apps in general. For example, here's [Page 1 ](https://itunes.apple.com/us/genre/ios-stickers/id6025?mt=8&letter=A)of 11 pages of Sticker packs in the store starting with the **letter A**.  

So maybe its a matter of the right packs catching their eyes now. Again, thats just speculation.

**Comment by ptkdev** [comment]
@cartoonsmart  Nope, i have 2 app. One is app for instagram stories, normal app with 10k download. The second is imessage stickers. 

I remove from sales imessage stickes, i create single app stickers+whatsapp but reviewer set at spam/design again. I don't know why...
